### PR TITLE
Clarify Intptr use in SID constructor

### PR DESCRIPTION
**Pull request by henke37**
It is a pointer and not some weird type overload.

### CODE REVIEW

**Comment by henke37** [comment]
@dotnet-policy-service agree

File: xml/System.Security.Principal/SecurityIdentifier.xml [hunk 1]
```diff
@@ -81,8 +81,8 @@
         <Parameter Name="binaryForm" Type="System.IntPtr" />
       </Parameters>
       <Docs>
-        <param name="binaryForm">An integer that represents the binary form of a SID.</param>
-        <summary>Initializes a new instance of the <see cref="T:System.Security.Principal.SecurityIdentifier" /> class by using an integer that represents the binary form of a security identifier (SID).</summary>
+        <param name="binaryForm">An integer that converts to a pointer to the binary form of a SID.</param>
+        <summary>Initializes a new instance of the <see cref="T:System.Security.Principal.SecurityIdentifier" /> class by using an integer that converts to a pointer to the binary form of a security identifier (SID).</summary>
```
Comment (adamsitnik) [inline xml/System.Security.Principal/SecurityIdentifier.xml:85]: I think that we can go one step further and just say that it's a pointer.

```suggestion
        <param name="binaryForm">A pointer to the binary form of a SID.</param>
        <summary>Initializes a new instance of the <see cref="T:System.Security.Principal.SecurityIdentifier" /> class by using a pointer to the binary form of a security identifier (SID).</summary>
```

**Review by adamsitnik — COMMENTED** [review]
@henke37 thank you for your contribution!

PTAL at my suggestion and let me know what do you think.

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you for improving our docs @henke37 !

**Pull request merged**